server/rest/src/main/java/org/infinispan/rest/NettyRestServer.java

```diff
@@ -70,30 +49,55 @@ private static EmbeddedCacheManager createCacheManager(String cfgFile) {
       }
    }
 
-   private NettyRestServer(EmbeddedCacheManager cacheManager, RestServerConfiguration configuration, NettyJaxrsServer netty,
-                           Consumer<? super EmbeddedCacheManager> onStop) {
+   private NettyRestServer(EmbeddedCacheManager cacheManager, RestServerConfiguration configuration) {
       this.cacheManager = cacheManager;
       this.configuration = configuration;
-      this.netty = netty;
-      this.onStop = onStop;
    }
 
    @Override
    public void start() {
-      netty.start();
-      ResteasyDeployment deployment = netty.getDeployment();
       configuration.getIgnoredCaches().forEach(this::ignoreCache);
       RestCacheManager restCacheManager = new RestCacheManager(cacheManager, this::isCacheIgnored);
-      Server server = new Server(configuration, restCacheManager);
-      deployment.getRegistry().addSingletonResource(server);
-      deployment.getProviderFactory().register(new RestAccessLoggingHandler(), ContainerRequestFilter.class,
-            ContainerResponseFilter.class);
-      log.startRestServer(configuration.host(), configuration.port());
+      server = Optional.of(new Server(configuration, restCacheManager));
+
+      if(configuration.startTransport()) {
+         NettyJaxrsServer nettyServer = createNetty();
+         nettyServer.start();
+         ResteasyDeployment deployment = nettyServer.getDeployment();
+         deployment.getRegistry().addSingletonResource(server.get(), DEFAULT_REST_PATH);
+         deployment.getProviderFactory().register(new RestAccessLoggingHandler(), ContainerRequestFilter.class,
+               ContainerResponseFilter.class);
+         log.startRestServer(configuration.host(), configuration.port());
+         netty = Optional.of(nettyServer);
+      }
+   }
+
+   private NettyJaxrsServer createNetty() {
+      // Start caches first, if not started
+      startCaches(cacheManager);
+
+      NettyJaxrsServer netty = new NettyJaxrsServer();
+      ResteasyDeployment deployment = new ResteasyDeployment();
+      netty.setDeployment(deployment);
+      netty.setHostname(configuration.host());
+      netty.setPort(configuration.port());
+      netty.setRootResourcePath("");
+      netty.setSecurityDomain(null);
+      return netty;
    }
 
    @Override
    public void stop() {
-      netty.stop();
-      onStop.accept(cacheManager);
+      netty.ifPresent(NettyJaxrsServer::stop);
+      netty = Optional.empty();
+      server = Optional.empty();
```

Comment:
I thought you'd need a separate `stop()` for the `Server`, but I see it doesn't even have a `stop()` method :)